86 pauseescape menu

Heldereim opened the pull request:
Added a pause menu to the game that is triggered by pressing the TAB button.
Pausing the game halts all game processes (unless specified otherwise). 

In the pause menu the player can press the resume button to continue the game.

Heldereim committed:
Pause menu

Added a pause menu that is triggered when pressing the TAB button. Added scripts for this functionality.

Heldereim committed:
Pause menu added to game

Added pause menu to the user interface scene. Moved the pause menu scripts to the UI scripts folder.

linuskar reviewed (changes requested):
- I feel the escape key is better to use for the pause menu to popup. 
- Would be good to also make they key for the pause menu be able to toggle the menu on and off when pressed. 
- Could you add a button that quits to desktop (quit game).

Heldereim committed:
Pause toggleable, exit game button

Made the pause key toggleable, also added a button in the pause menu for quitting the game

Heldereim committed:
Pause button to game header

Heldereim committed:
UIMenu

Made a script for a class called UIMenu that is for UIElements that are also menus. Made it so that pausing the game happens on escape instead of tab. Added code that closes menus when pressing escape instead of pausing the game when menus are opened.

Simbpa committed:
Added game paused label

Simbpa reviewed (approved):
Lgtm
